mail_tracking/models/mail_tracking_email.py

```diff
@@ -77,13 +84,99 @@ class MailTrackingEmail(models.Model):
         string="Tracking events", comodel_name='mail.tracking.event',
         inverse_name='tracking_email_id', readonly=True)
 
+    @api.model
+    def tracking_ids_recalculate(self, model, email_field, tracking_field,
+                                 email, new_tracking=None):
+        objects = self.env[model].search([
+            (email_field, '=ilike', email),
+        ])
+        for obj in objects:
+            trackings = obj[tracking_field]
+            if new_tracking:
+                trackings |= new_tracking
+            trackings = trackings._email_score_tracking_filter()
+            if set(obj[tracking_field].ids) != set(trackings.ids):
+                if trackings:
+                    obj.write({
+                        tracking_field: [(6, False, trackings.ids)]
+                    })
+                else:
+                    obj.write({
+                        tracking_field: [(5, False, False)]
+                    })
+        return True
+
+    @api.model
+    def _tracking_ids_to_write(self, email):
+        trackings = self.env['mail.tracking.email'].search([
+            ('recipient_address', '=ilike', email)
+        ])
+        trackings = trackings._email_score_tracking_filter()
+        if trackings:
+            return [(6, False, trackings.ids)]
+        else:
+            return [(5, False, False)]
+
+    @api.multi
+    def _email_score_tracking_filter(self):
+        """Default email score filter for tracking emails"""
+        # Consider only last 10 tracking emails
+        return self.sorted(key=lambda r: r.time, reverse=True)[:10]
+
+    @api.multi
+    def email_score(self):
+        """Default email score algorimth"""
+        score = 50.0
+        trackings = self._email_score_tracking_filter()
+        for tracking in trackings:
+            if tracking.state in ('error',):
+                score -= 50.0
+            elif tracking.state in ('rejected', 'spam', 'bounced'):
+                score -= 25.0
+            elif tracking.state in ('soft-bounced', 'unsub'):
+                score -= 10.0
+            elif tracking.state in ('delivered',):
+                score += 5.0
+            elif tracking.state in ('opened',):
+                score += 10.0
+        if score > 100.0:
+            score = 100.0
+        return score
+
+    @api.multi
+    @api.depends('recipient')
+    def _compute_recipient_address(self):
+        for email in self:
+            matches = re.search(r'<(.*@.*)>', email.recipient)
```

Comment:
@hbrunn This only could happend if partner email is ```hello@world>``` because this form if built by Odoo automatically concatenating partner name and partner email.
This match is not for checking, just for extracting and easy matching by partner email afterwords